### ISSUE TITLE
"Could not load assembly 'Test'. Ensure it is referenced by the startup project 'Test'" error while adding migration for x86 .NET 5.0-preview7 project with dotnet ef 5.0-rc1

### ISSUE THREAD

**Issue by bairog** [dotnet/efcore#22594]
Hello.
I have simple class library project with a EF Core 5.0-rc1 DbContext that targets .NET 5.0-preview7 and is build in **x86** configuration. I'm using **.NET 5-preview7** because according to [info on download page](https://dotnet.microsoft.com/download/dotnet/5.0) it is the latest one that supports Visual Studio 2019 v**16.7 Release**. I use only stable IDE builds in my project and cannot use preview IDE versions.
I've installed latest Entity Framework Core .NET Command-line Tools 5.0.0-rc.1.20451.13
When I'm trying to add migration via **dotnet ef migrations add InitialCreate --verbose** command I get an error:
```
Could not load assembly 'Test'. Ensure it is referenced by the startup project 'Test'.
```

### Steps to reproduce

1. Download test project - https://yadi.sk/d/NvL-nyrwnTgrcA
2. Install latest Entity Framework Core .NET Command-line Tools 5.0.0-rc.1.20451.13
3. Run **dotnet ef migrations add InitialCreate --verbose** command in test project folder

**P. S.** I've tried to separate Entites from DbContext and created separate class library for Entities. 
I've changed DbContext project configuration to **AnyCPU** but left Entites project configuration to **x86**.
And now **dotnet ef migrations add InitialCreate --verbose** results in error
```
Unable to create an object of type 'BlankDBContext'. For the different patterns supported at design time, see https://go.microsoft.com/fwlink/?linkid=851728
```
Here is test project with Entites is separate project - https://yadi.sk/d/qLApIBTKfRhyMA

Looks like the is broad problem with **x86** configuration.

### dotnet ef migrations add InitialCreate --verbose output:

```
D:\Test\Test>dotnet ef migrations add InitialCreate --verbose
Using project 'D:\Test\Test\Test.csproj'.
Using startup project 'D:\Test\Test\Test.csproj'.
Writing 'D:\Test\Test\obj\Test.csproj.EntityFrameworkCore.targets'...
dotnet msbuild /target:GetEFProjectMetadata /property:EFProjectMetadataFile=C:\Users\Leo.TOPAZ1\AppData\Local\Temp\tmpEC82.tmp
 /verbosity:quiet /nologo D:\Test\Test\Test.csproj
Writing 'D:\Test\Test\obj\Test.csproj.EntityFrameworkCore.targets'...
dotnet msbuild /target:GetEFProjectMetadata /property:EFProjectMetadataFile=C:\Users\Leo.TOPAZ1\AppData\Local\Temp\tmpEE67.tmp
 /verbosity:quiet /nologo D:\Test\Test\Test.csproj
Build started...
dotnet build D:\Test\Test\Test.csproj /verbosity:quiet /nologo

Сборка успешно завершена.
    Предупреждений: 0
    Ошибок: 0

Прошло времени 00:00:00.84
Build succeeded.
dotnet exec --depsfile D:\Test\Test\bin\Debug\net5.0\Test.deps.json --additionalprobingpath C:\Users\Leo.TOPAZ1\.nuget\packages
 --additionalprobingpath "C:\Program Files\dotnet\sdk\NuGetFallbackFolder" --runtimeconfig 
D:\Test\Test\bin\Debug\net5.0\Test.runtimeconfig.json
 C:\Users\Leo.TOPAZ1\.dotnet\tools\.store\dotnet-ef\5.0.0-rc.1.20451.13\dotnet-ef\5.0.0-rc.1.20451.13\tools\netcoreapp3.1\any\tools\netcoreapp2.0\any\ef.dll
 migrations add InitialCreate --assembly D:\Test\Test\bin\Debug\net5.0\Test.dll
 --startup-assembly D:\Test\Test\bin\Debug\net5.0\Test.dll --project-dir D:\Test\Test\ --language C# --working-dir D:\Test\Test --verbose --root-namespace Test
Using assembly 'Test'.
Using startup assembly 'Test'.
Using application base 'D:\Test\Test\bin\Debug\net5.0'.
Using working directory 'D:\Test\Test'.
Using root namespace 'Test'.
Using project directory 'D:\Test\Test\'.
Remaining arguments: .
Microsoft.EntityFrameworkCore.Design.OperationException: Could not load assembly 'Test'. Ensure it is referenced by the startup project 'Test'.
 ---> System.BadImageFormatException: Could not load file or assembly 'Test, Culture=neutral, PublicKeyToken=null'. Была сделана попытка загрузить программу, имеющую неверный формат.
File name: 'Test, Culture=neutral, PublicKeyToken=null'
   at System.Reflection.RuntimeAssembly.InternalLoad(ObjectHandleOnStack assemblyName, ObjectHandleOnStack requestingAssembly, StackCrawlMarkHandle stackMark, Boolean throwOnFileNotFound, ObjectHandleOnStack assemblyLoadContext, ObjectHandleOnStack retAssembly)
   at System.Reflection.RuntimeAssembly.InternalLoad(AssemblyName assemblyName, RuntimeAssembly requestingAssembly, StackCrawlMark& stackMark, Boolean throwOnFileNotFound, AssemblyLoadContext assemblyLoadContext)
   at System.Reflection.Assembly.Load(AssemblyName assemblyRef)
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.<get_Assembly>g__Create|14_0()
   --- End of inner exception stack trace ---
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.<get_Assembly>g__Create|14_0()
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.get_Assembly()
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.get_MigrationsOperations()
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.AddMigrationImpl(String name, String outputDir, String contextType, String namespace)
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.AddMigration.<>c__DisplayClass0_0.<.ctor>b__0()
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.OperationBase.<>c__DisplayClass3_0`1.<Execute>b__0()
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.OperationBase.Execute(Action action)
Could not load assembly 'Test'. Ensure it is referenced by the startup project 'Test'.
```

### Further technical details

EF Core version: **5.0-rc1**
Database provider: **Microsoft.EntityFrameworkCore.Sqlite 5.0-rc1**
Target framework: **.NET 5.0-preview7**
Operating system: **Windows 10 2004 x64**
IDE: **Visual Studio 2019 16.7.3 Professional**

**Comment by ajcvickers** [comment]
See also #22593

**Comment by ajcvickers** [comment]
@bricelam This repros with the RC1 SDK. Likely the issue with x86 configuration that you mentioned in triage.

**Comment by bricelam** [comment]
Loosely related to the discussion in #20752. Possibly solved by #14225.

It should work if the 32-bit SDK is installed and first on your path.

The basic problem here is that you'd get the same error if you run the following command using the 64-bit SDK/runtime. The 64-bit dotnet.exe tries to load the 32-bit Test.dll.
```
dotnet bin\Debug\net5.0\Test.dll
```
I'll have to dig into the details (and probably work with the cli team) to figure out how to find the 32-bit version of dotnet.exe. This is not a new issue--it's been this way since the beginning.

**Comment by bairog** [comment]
@bricelam @ajcvickers The problem reproduces on **.NET 5.0.2** project with **dotnet ef 5.0.2**. 
Any progress for now?

**Comment by luojunyuan** [comment]
currently  I use `"C:\Program Files (x86)\dotnet\dotnet.exe" ef migrations add InitialCreate` to run the command

**Comment by t13801206** [comment]
I got the same error message, and solved by #14225
1. Install .NET 5.0 **x86** SDK
2. Add SDK to %PATH%

I moved `C:\Program Files (x86)\dotnet\` above ``C:\Program Files\dotnet\``.  (maybe you need to restart process)
```cmd
>echo %PATH%
C:\Program Files (x86)\dotnet\;C:\Program Files\dotnet\;
```

But this time, I got the same error **on x64 build** 😭

**Comment by ajcvickers** [comment]
Note for triage: verified that if both the x64 and x86 SDKs are installed, then the x86 SDK needs to be first on the path for this to work, and conversely, the x64 version must be first on the path to use projects targeted specifically at x64. As Brice pointed out, this is the same behavior as when launching the DLL directly.

Not sure that EF should do anything specific here. We should probably avoid adding more magic to try to discover anything, given that kind of magic has a tendency to break in future versions.
